lib/src/sidenav/types.ts

```diff
@@ -0,0 +1,55 @@
+type Space = "xxsmall" | "xsmall" | "small" | "medium" | "large" | "xlarge" | "xxlarge";
+
+type Padding = {
+  top?: Space;
+  bottom?: Space;
+  left?: Space;
+  right?: Space;
+};
+
+export type SidenavLinkPropsType = {
+  /**
+   * Value of the tabindex.
+   */
+  tabIndex?: number;
+  /**
+   * Page to be opened when the user clicks on the link.
+   */
+  href: string;
+  /**
+   * This function will be called when the user clicks the link.
+   */
+  onClick: () => void;
```

Comment:
I think the `onClick` event should be optional. Only href and children are really necessary for rendering correctly the component.

Reply:
I think this is a similar case to the link. You can have either onClick or href, both are optional but at least one should be necessary.